cli/command/service/progress/progress.go

```diff
@@ -303,7 +327,23 @@ func (u *replicatedProgressUpdater) update(service swarm.Service, tasks []swarm.
 			u.slotMap[task.Slot] = mappedSlot
 		}
 
-		if !u.done && replicas <= maxProgressBars && uint64(mappedSlot) <= replicas {
+		if !terminalState(task.DesiredState) && task.Status.State == swarm.TaskStateRunning {
+			running++
+		}
```

Comment:
Could the body of this for loop be a separate method which returns the running count?

I think that would improve the readability of this function, and would make it easier to test this new code.